[JENKINS-70558] Fix TopicAssociation not set after Jenkins restart

ckreisl opened the pull request:
<!-- Please describe your pull request here. -->

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

The PR #481 introduced a small bug that the `TopicAssociation` field is not set after a Jenkins restart.

Description from Jenkins issue: https://issues.jenkins.io/browse/JENKINS-70558 

In the job configuration if TopicAssociation is enabled and Jenkins is restarted the option is not set anymore.
The reason is that the `readResolve()` function checks the config value 
`private transient boolean enableTopicAssociation = Config.DEFAULT_ENABLE_TOPIC_ASSOCIATION;`
 (GerritTrigger.java) which is always **false**. Since this condition is always the case the topicAssociation value in the GerritTrigger class is set to "null".

ckreisl commented on src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java:
```diff
@@ -1893,10 +1893,12 @@ public void setSilentMode(boolean silentMode) {
      * @param enable true or false.
      */
     @Deprecated
+    @DataBoundSetter
```
I think that's the way how we want it. If the old build configuration xml is read it will set already create the new TopicAssociation class.

Ich think line 2181 becomes then obsolete?

rsandell replied:
`config.xml` deserialization doesn't care about `@DataBoundSetter` et.al. Those are only relevant for form submission and CasC.

rsandell reviewed (changes requested):
You can write a test using [JenkinsSessionRule](https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/JenkinsSessionRule.html) or [RealJenkinsRule](https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/RealJenkinsRule.html) to test that the configuration actually survives a restart.

ckreisl commented:
> You can write a test using [JenkinsSessionRule](https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/JenkinsSessionRule.html) or [RealJenkinsRule](https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/RealJenkinsRule.html) to test that the configuration actually survives a restart.

Thanks for the pointer, added a test now and removed the `@DataBoundSetter`.
